packages/schematics/angular/migrations/update-9/update-app-tsconfigs.ts

```diff
@@ -110,12 +110,12 @@ function updateTsConfig(tree: Tree, builderConfig: JsonAstObject, builderName: B
 
         const mainOption = findPropertyInAstObject(option, 'main');
         if (mainOption && mainOption.kind === 'string') {
-          newFiles.push(posix.relative(posix.dirname(tsConfigPath), mainOption.value));
+          newFiles.push(relative(dirname(tsConfigPath), mainOption.value).replace(/\\/g, '/'));
```

Comment:
Posix resolution on Windows will still prefer the windows path separator: https://nodejs.org/api/path.html#path_path_normalize_path

But here what we really want is a "TS-style" separator instead, which is forward slash everywhere.

Reply:
Also unfortunate is that this code:
```
const tsConfigPath = 'projects\\another-app\\tsconfig.app.json';
console.log(dirname(tsConfigPath));
console.log(posix.dirname(tsConfigPath));
```

Will print `.` twice on Linux, but on Windows will print `projects\another-app` and `.`. So I don't think the posix functions help us at all here.